packages/@dcl/inspector/src/components/Tree/Tree.tsx

```diff
@@ -115,41 +118,50 @@ function Tree<T>(props: Props<T>) {
     onRemove(value)
   }
 
+  const ref = (node: HTMLDivElement | null) => drag(drop(node))
+
+  const controlsProps = {
+    enableAdd: enableAddChild,
+    enableEdit: enableRename,
+    enableRemove,
+    handleEdit: handleToggleEdit,
+    handleAdd: handleNewChild,
+    handleRemove: handleRemove
+  }
+
+  return (
+    <div ref={ref} className="main">
+      <div style={getLevelStyles(level)} className={selected ? 'selected' : ''}>
+        <span onClick={handleToggleExpand} style={getEditModeStyles(editMode)}>
+          <Arrow isOpen={open} />
+          <span>{label || id}</span>
+        </span>
+        {editMode && <Input value={label || ''} onCancel={quitEditMode} onSubmit={onChangeEditValue} />}
+        <Controls {...controlsProps} />
+      </div>
+      <TreeChildren {...props} />
+      {insertMode && <Input value="" onCancel={quitInsertMode} onSubmit={handleAddChild} />}
+    </div>
+  )
+}
+
+function Arrow({ isOpen }: { isOpen: boolean }) {
+  return <>{isOpen ? <ArrowDown /> : <ArrowRight />}</>
+}
+
+function TreeChildren<T>(props: Props<T>) {
```

Comment:
Why moving this part into a new component for this instead of doing it within `Tree`?

Reply:
the `Tree` component already has a lot of logic in it. It reads better by removing all the `{...logic}` stuff from it and just leave markup with props in the main component